pkg/graph/schema/types/translation.graphql

```diff
@@ -0,0 +1,294 @@
+  """
+  parentReferencesLabel: String
+  """
+  Labels specifically for export/change history.  Takes priority over all other labels
+  """
+  exportLabel: String
+  """
+  Table reference for fields that are of dataType UUID and reference a table in the database
+  """
+  tableReference: TableName
+}
+
+"""
+Represents a translation question with options
+"""
+type TranslationFieldWithOptions {
+  gqlField: String!
+  goField: String!
+  dbField: String!
+  label: String!
+  readonlyLabel: String
+  sublabel: String
+  multiSelectLabel: String
+  isArray: Boolean
+  dataType: TranslationDataType!
+  formType: TranslationFormType!
+  isNote: Boolean
+  """
+  Designates the order of the question in the form.  Uses integer as page and question order uses hundreths place.  Ex: 1.01, 1.02, 2.01, 2.02
+  """
+  order: Float!
+  """
+  Is a question a followup to another that doesn't designate it's own readonly question/line
+  """
+  isOtherType: Boolean
+  """
+  Field name for the parent question for fields that represent Other, Please specify, etc.  Used in change history to render parent question for context
+  """
+  otherParentField: String
+  """
+  Label for fields that reference more than one parent - Ex: Notes - 'Note for Model Basics'
+  """
+  parentReferencesLabel: String
+  """
+  Labels specifically for export/change history.  Takes priority over all other labels
+  """
+  exportLabel: String
+  """
+  Table reference for fields that are of dataType UUID and reference a table in the database
+  """
+  tableReference: TableName
+  options: Map!
+  exportOptions: Map
+}
+
+"""
+Represents a translation question with no options and a parent
+"""
+type TranslationFieldWithParent {
+  gqlField: String!
+  goField: String!
+  dbField: String!
+  label: String!
+  readonlyLabel: String
+  sublabel: String
+  multiSelectLabel: String
+  isArray: Boolean
+  dataType: TranslationDataType!
+  formType: TranslationFormType!
+  isNote: Boolean
+  """
+  Designates the order of the question in the form.  Uses integer as page and question order uses hundreths place.  Ex: 1.01, 1.02, 2.01, 2.02
+  """
+  order: Float!
+  """
+  Is a question a followup to another that doesn't designate it's own readonly question/line
+  """
+  isOtherType: Boolean
+  """
+  Field name for the parent question for fields that represent Other, Please specify, etc.  Used in change history to render parent question for context
+  """
+  otherParentField: String
+  """
+  Label for fields that reference more than one parent - Ex: Notes - 'Note for Model Basics'
+  """
+  parentReferencesLabel: String
+  """
+  Labels specifically for export/change history.  Takes priority over all other labels
+  """
+  exportLabel: String
+  """
+  Table reference for fields that are of dataType UUID and reference a table in the database
+  """
+  tableReference: TableName
+  parentRelation: TranslationField!
+}
```

Comment:
There may be an approach to make this code a bit more DRY by implementing an interface, that way you're at least not duplicating documentation:

```"""
Interface representing common fields for translation questions
"""
interface TranslationFieldBase {
  gqlField: String!
  goField: String!
  dbField: String!
  label: String!
  readonlyLabel: String
  sublabel: String
  multiSelectLabel: String
  isArray: Boolean
  dataType: TranslationDataType!
  formType: TranslationFormType!
  isNote: Boolean
  """
  Designates the order of the question in the form. Uses integer as page and question order uses hundredths place. Ex: 1.01, 1.02, 2.01, 2.02
  """
  order: Float!
  """
  Is a question a followup to another that doesn't designate its own readonly question/line
  """
  isOtherType: Boolean
  """
  Field name for the parent question for fields that represent Other, Please specify, etc. Used in change history to render parent question for context
  """
  otherParentField: String
  """
  Label for fields that reference more than one parent - Ex: Notes - 'Note for Model Basics'
  """
  parentReferencesLabel: String
  """
  Labels specifically for export/change history. Takes priority over all other labels
  """
  exportLabel: String
  """
  Table reference for fields that are of dataType UUID and reference a table in the database
  """
  tableReference: TableName
}

"""
Represents a translation question with no options
"""
type TranslationField implements TranslationFieldBase {
  gqlField: String!
  goField: String!
  dbField: String!
  label: String!
  readonlyLabel: String
  sublabel: String
  multiSelectLabel: String
  isArray: Boolean
  dataType: TranslationDataType!
  formType: TranslationFormType!
  isNote: Boolean
  order: Float!
  isOtherType: Boolean
  otherParentField: String
  parentReferencesLabel: String
  exportLabel: String
  tableReference: TableName
}

"""
Represents a translation question with options
"""
type TranslationFieldWithOptions implements TranslationFieldBase {
  gqlField: String!
  goField: String!
  dbField: String!
  label: String!
  readonlyLabel: String
  sublabel: String
  multiSelectLabel: String
  isArray: Boolean
  dataType: TranslationDataType!
  formType: TranslationFormType!
  isNote: Boolean
  order: Float!
  isOtherType: Boolean
  otherParentField: String
  parentReferencesLabel: String
  exportLabel: String
  tableReference: TableName
  options: Map!
  exportOptions: Map
}```

Reply:
Thanks Tom! Interesting idea, I'll test it out and see if it adds a lot of complexity. My goal here is to keep documentation on each type when the schema is queried with introspection.

Reply:
I do like that it would have a common place for documentation, but testing this out reveals a lot of extra wiring is required to make this work with GQLgen. 

GQL gen defaults to generating a getter per field. I was able to disable that behavior with configuration. When I tested out Implementing interface on the backend type, and then doing the introspection query in graphql, the types that implement the interface don't inherit the documentation comment from graphql. 

From my research and experimentation, it seems that if you want to have documentation on a type in graphql, you have to define it individually on each field. I think it would be good to leave these with duplicated comments as this seems to be how graphql functions.

Reply:
Yes. I don't think there's particularly DRY solution for this.